### PR TITLE
DataTexture: Added serialization/deserialization support.

### PR DESCRIPTION
**Pull request by Mugen87**
This PR ensures `DataTexture` is now full serializable/deserializable. That means:

- `.toJSON()` does not break anymore when instances of `DataTexture` are used in the scene (see https://jsfiddle.net/ba29mgnj/)
- `three.js` dev tools support now scenes with `DataTexture`
- The PR ensures `DataTexture` can be serialized/deserialized when used as single textures or when part of a cube texture.

Besides, the PR makes the texture code in `ObjectLoader` a bit more robust. `needsUpdate` is only set if texture data are actually available/complete.

BTW: This PR is a prerequisite to enable HDR textures in the editor. Right now, the editor produces runtime errors when HDR textures are applied to materials.

### CODE REVIEW

**Comment by Mugen87** [comment]
BTW: This change would make #16816 obsolete since `ImageUtils.getDataURL` is now only called with valid input.

**Comment by mikelattanziobluescape** [comment]
Good morning everyone.

Ran into this issue with my team here just recently and found this PR - anything preventing this bad boy from landing in the near future?

File: src/utils.js [hunk 1]
```diff
@@ -30,4 +30,23 @@ function arrayMax( array ) {
...
+	Uint8ClampedArray: typeof Uint8ClampedArray !== 'undefined' ? Uint8ClampedArray : Uint8Array,
+	Int16Array: Int16Array,
+	Uint16Array: Uint16Array,
+	Int32Array: Int32Array,
+	Uint32Array: Uint32Array,
+	Float32Array: Float32Array,
+	Float64Array: Float64Array
+};
+
+function getTypedArray( type, buffer ) {
```
Comment (Mugen87) [inline src/utils.js:46]: Finally found a way to avoid code duplicate in context of typed array creation^^.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**